### PR TITLE
pfserving tracing improvemnt

### PR DESCRIPTION
**Pull request by wxpjimmy**
# Description
- support disable built-in trace exporters
- force disable local setup exporter logic for serving mode with "OTEL_EXPORTER_OTLP_ENDPOINT"

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by zhengfeiwang — DISMISSED** [review]

**Review by brynn-code — APPROVED** [review]

**Review by huaiyan — APPROVED** [review]

File: src/promptflow-tracing/promptflow/tracing/_start_trace.py [hunk 1]
```diff
@@ -108,12 +108,17 @@ def start_trace(
             start_trace_with_devkit(collection=collection, attributes=_attributes, run=_run)
 
 
-def setup_exporter_from_environ() -> None:
+def setup_exporter_from_environ(only_inject_openai: bool = False) -> None:
     # openai instrumentation
     logging.debug("injecting OpenAI API...")
     inject_openai_api()
```
Comment (wangchao1230) [inline src/promptflow-tracing/promptflow/tracing/_start_trace.py:114]: shall we move the inject_oepn_api out of the setup_exporter_from_environ function? 
So we don't need to add this strance bool parameter here

File: src/promptflow-core/promptflow/core/_serving/constants.py [hunk 2]
```diff
@@ -4,3 +4,5 @@
 
 FEEDBACK_TRACE_FIELD_NAME = "feedback"
 FEEDBACK_TRACE_SPAN_NAME = "promptflow-feedback"
+
+PF_BUILTIN_TRACE_EXPORTERS_DISABLE = "PF_BUILTIN_TRACE_EXPORTERS_DISABLE"
```
Comment (thy09) [inline src/promptflow-core/promptflow/core/_serving/constants.py:8]: Why don't we use the env PF_TRACING_SKIP_LOCAL_SETUP_ENVIRON ?

Comment (wxpjimmy) [reply src/promptflow-core/promptflow/core/_serving/constants.py:8]: Purpose is different and this naming "PF_TRACING_SKIP_LOCAL_SETUP_ENVIRON " is confusing for customer. Here what they need is to disable the deployed endpoint's trace exporters even if there are APPLICATIONINSIGHTS_CONNECTION_STRING or OTEL_EXPORTER_OTLP_ENDPOINT configured. A separate name will make things more clear and easy to understand.

File: src/promptflow-tracing/promptflow/tracing/_start_trace.py (hunk 1 above)
Comment (wxpjimmy) [reply src/promptflow-tracing/promptflow/tracing/_start_trace.py:114]: That also makes sense to me,  @thy09  @huaiyan any comments on Clement's feedback?

File: src/promptflow-core/promptflow/core/_serving/constants.py (hunk 2 above)
Comment (wxpjimmy) [reply src/promptflow-core/promptflow/core/_serving/constants.py:8]: If you mean using PF_TRACING_SKIP_LOCAL_SETUP_ENVIRON to disable the setup, this "PF_TRACING_SKIP_LOCAL_SETUP_ENVIRON" doesn't work for setup_exporter_from_environ.

Comment (thy09) [reply src/promptflow-core/promptflow/core/_serving/constants.py:8]: I see, so the exporter would also be different here.

File: src/promptflow-tracing/promptflow/tracing/_start_trace.py (hunk 1 above)
Comment (wxpjimmy) [reply src/promptflow-tracing/promptflow/tracing/_start_trace.py]: updated based on comments

**Review by wangchao1230 — APPROVED** [review]

File: src/promptflow-core/promptflow/executor/flow_executor.py [hunk 3]
```diff
@@ -209,7 +209,11 @@ def create(
         :return: A new instance of FlowExecutor.
         :rtype: ~promptflow.executor.flow_executor.FlowExecutor
         """
-        setup_exporter_from_environ()
+        only_inject_openai = kwargs.get("only_inject_openai", False)
+        if only_inject_openai:
+            inject_openai_api()
+        else:
+            setup_exporter_from_environ()
```
Comment (brynn-code) [inline src/promptflow-core/promptflow/executor/flow_executor.py:216]: shall this branch do: 1. inject openai  2 set exporter?

Comment (wxpjimmy) [reply src/promptflow-core/promptflow/executor/flow_executor.py:214]: we didn't modify the existing func, it already calls inject

**Review by huaiyan — APPROVED** [review]

**Review by brynn-code — APPROVED** [review]

**Pull request merged**